File: code-examples/examples/example-3/src/main/java/be/swsb/typesafety/example3/external/legacybookingsystem/LegacyMapper.java
package be.swsb.typesafety.example3.external.legacybookingsystem;

import be.swsb.typesafety.example3.domain.booking.Booking;
import be.swsb.typesafety.example3.external.legacybookingsystem.acl.types.LegacyBookingType;

public class LegacyMapper {

    public LegacyBookingType toXml(final Booking booking) {
        final LegacyBookingType legacyBookingType = new LegacyBookingType();
        legacyBookingType.setCode(booking.getBookingCodeAsString());
        return legacyBookingType;
    }
}

File: code-examples/examples/example-5/src/main/java/be/swsb/typesafety/example5/external/legacybookingsystem/LegacyMapper.kt
package be.swsb.typesafety.example5.external.legacybookingsystem

import be.swsb.typesafety.example5.domain.booking.Booking
import be.swsb.typesafety.example5.external.legacybookingsystem.acl.types.LegacyBookingType

open class LegacyMapper {

    open fun toXml(booking: Booking): LegacyBookingType {
        val legacyBookingType = LegacyBookingType()
        legacyBookingType.setCode(booking.getBookingCodeAsString())
        return legacyBookingType
    }
}
